### DISCUSSION TITLE
Teleport Session Uploaded Issue

### DISCUSSION THREAD

**Discussion by mdsketch** [gravitational/teleport#11523] [Q&A] [answered]
I have a problem and I'm not sure if its an issue I have alone or if its a Bug.

Been using teleport for 8-10 months now with 20-40 IoT devices and a single proxy/auth server.

My issue is that I get a session upload in the audit log every 5-10ms all day every day. Every single event has the exact same session ID. When I try to view the recording with tsh I get this.
![image](https://user-images.githubusercontent.com/59035262/160478770-fb709576-514b-4bd1-85d7-1c51d5279b41.png)

And here are all the session logs:
![image](https://user-images.githubusercontent.com/59035262/160478876-64777e99-b59c-4973-844d-1ec8660617aa.png)
![image](https://user-images.githubusercontent.com/59035262/160479012-795f0aed-dfc6-4c19-b847-48ae3d98259b.png)

This started about 3 months ago and I noticed because teleport would take 100% CPU up from the 5-10% it usually does. A reboot fixed it but then about 1 month later back to 100%. This was on version 7.2.1, am now on 8.3.4 (can't go any newer until I update my nodes).

When the server was at 100% no users noticed any issues with speed either, digging in it looks like it was high usage because of the volume of logs being written.

I'm really not sure what else to do. I cant restart from scratch and this is more an annoyance at the current stage, but I want to catch it before it morphs to something else.

Here is the ever repeating log message
![image](https://user-images.githubusercontent.com/59035262/160479710-d54a06b6-3d9d-4d74-9ce3-05baed63d6bd.png)

**Comment by zmb3** [comment]
Hey @mdsketch - I'd like to help you with this.

What are you using for session storage? (`teleport.storage.audit_events_uri` in your `teleport.yaml`)

What is your `session_recording` mode?

**Reply by mdsketch** [reply]
`session_recording: "node"`
and I am just using the default storage configuration.

Another thing that popped up since I posed this is the same sort of ghost issue for applications. I removed a node (app and node service) last week yet it still shows up in the applications page. If I click on it the page goes into a redirect loop. I haven't had the opportunity to schedule downtime to restart the cluster yet.

**Reply by zmb3** [reply]
The app issue seems like a separate thing. That should definitely disappear some time after the app service stops heartbeating.

For now I'd like to keep focused on the session upload issue.

Is there a `/var/lib/teleport/log/records/multi/<SESSION_ID>` directory on the node for one of the bad session IDs? If so, what's inside it? Is it empty or are there part files?

**Reply by mdsketch** [reply]
So the bad session ID isn't a directory in `/var/lib/teleport/log/records/multi/<BAD_SESSION_ID>`

So I ran `find -type d -name "<BAD_SESSION_ID>"` and as far as I can tell I got a directory named the bad session id as a sub directory in most if not all the directories in `/var/lib/teleport/log/records/multi/<SOME_SID>/<BAD_SESSION_ID>`

I went into a few of them, they look empty. Also all the directories in `/var/lib/teleport/log/records/multi/<SOME_SID>` were created today.

**Reply by zmb3** [reply]
Would you be interested in setting up a time to troubleshoot with me via Slack and/or video? If so, let's try to connect in the [Teleport Community Slack](https://goteleport.slack.com/join/shared_invite/zt-midnn9bn-AQKcq5NNDs9ojELKlgwJUA#/shared-invite/email) and organize a time.

**Comment by zmb3** [comment, marked as answer]
This ended up being due to the a node failing to upload sessions because it hit the open file descriptor limit, as mentioned in https://github.com/gravitational/teleport/issues/10660.

This issue was fixed in v9.0.0, v8.3.4, and 7.3.18.
